src/stackdriver-nozzle/firehose/client.go

```diff
@@ -0,0 +1,54 @@
+package firehose
+
+import (
+	"github.com/cloudfoundry-community/firehose-to-syslog/caching"
+	"github.com/cloudfoundry-community/firehose-to-syslog/eventRouting"
+	"github.com/cloudfoundry-community/firehose-to-syslog/firehoseclient"
+	"github.com/cloudfoundry-community/firehose-to-syslog/logging"
+	"github.com/cloudfoundry-community/go-cfclient"
+)
+
+type Client interface {
+	StartListening(nozzle logging.Logging) error
+}
+
+type client struct {
+	cfConfig *cfclient.Config
+}
+
+func NewClient(apiAddress, username, password string, skipSSLValidation bool) Client {
+	return &client{
+		&cfclient.Config{
+			ApiAddress:        apiAddress,
+			Username:          username,
+			Password:          password,
+			SkipSslValidation: skipSSLValidation}}
+}
+
+func (fc *client) StartListening(nozzle logging.Logging) error {
+	cfClient := cfclient.NewClient(fc.cfConfig)
+
+	cachingClient := caching.NewCachingEmpty()
+
+	eventRouter := eventRouting.NewEventRouting(cachingClient, nozzle)
+
+	//err := eventRouter.SetupEventRouting("LogMessage,ValueMetric,HttpStartStop,CounterEvent,Error,ContainerMetric")
```

Comment:
It doesn't look like it from the source, but we can create the full list ourselves from here:

https://github.com/cloudfoundry/sonde-go/blob/master/events/envelope.pb.go#L61-L76

When we implement the config for filtering, it should become moot, I think?